File: 2025/day2/q1.py
sum = 0

with open("q1.txt") as f:
    line = f.readline().strip()
    ranges = line.split(",")
    for val in ranges:
        elems = val.split("-")
        s = int(elems[0])
        e = int(elems[1])
        
        # print(s, e)
        
        for i in range(s, e+1):
            st = str(i)
            n = len(st)
            if n % 2: continue
            # print(i, st[:n//2], st[n//2:])
            if st[:n//2] == st[n//2:]:
                print(i)
                sum += i
        
        
ranges = []


print("Sum: ", sum)

File: 2025/day2/q1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

int main() {
    // Open the file
    std::ifstream inputFile("test.txt");
    if (!inputFile) {
        std::cerr << "Error: Unable to open q1.txt" << std::endl;
        return 1;
    }

    std::vector<std::vector<char>> data;
    std::string line;

    while (std::getline(inputFile, line)) {
    }

    // Output the result
    std::cout << "Result: " << password << std::endl;

    return 0;
}